UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareServiceCache.cs

```diff
@@ -121,7 +124,17 @@ private Task<BatchDetail> CheckIfCacheProductsExistsInBlob(string exchangeSetRoo
             {
                 var bundleLocation = item.Bundle.FirstOrDefault().Location.Split(";");
                 exchangeSetRootPath = string.Format(exchangeSetRootPath, bundleLocation[0].Substring(1, 1), bundleLocation[1]);
-                downloadPath = Path.Combine(exchangeSetRootPath, item.ProductName.Substring(0, StringLength), item.ProductName, item.EditionNumber.Value.ToString());
+                if (aioConfiguration.IsAioEnabled)
```

Comment:
Please split out logic into a seperate method, as this method is becoming very large